### PR TITLE
fix: fix MsSQL JDBC URL format error 

### LINKED ISSUE #22998: [Bug]: MsSQL: connection string is rejected when trying to connect to Azure SQL server instance.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Error: `Driver.microsoft.sqlserver.jdbc.SQLServerDriver claims to not accept jdbcUrl`
Ref: https://theappsmith.slack.com/archives/C0341RERY4R/p1683087333331029 

### Steps To Reproduce

https://theappsmith.slack.com/archives/C0341RERY4R/p1683087333331029

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

self hosted, cloud

Front logo [Front conversations](https://app.frontapp.com/open/top_9w2oc)

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- MSSQL driver is not able to parse the JDBC URL properly if a semicolon is provided to it as part of the password and the password is added directly to the URL. 
  - This PR updates the way the password gets added. Instead of adding the password to the URL, this PR adds the password to HikariCP config. 
  - Please note that due to some misunderstanding both the ways of adding the password were present in the code, hence this PR only removes the code that adds password directly to URL. The code to add password to HikariCP config was already present, hence no code is added for it. 

#### PR fixes following issue(s)
Fixes #22998 

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [x] JUnit

#### How to test
- Create a user account in MsSQL DB with password that contains a semicolon in the end like `Password;3`. 
- Try to create and run query against the above account. 

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-plugins/mssqlPlugin/src/main/java/com/external/plugins/MssqlPlugin.java [hunk 1]
```diff
@@ -516,20 +516,6 @@ private static HikariDataSource createConnectionPool(DatasourceConfiguration dat
                     .append(";");
         }
 
-        if (StringUtils.hasLength(authentication.getUsername())) {
```
Comment (sumitsum) [inline app/server/appsmith-plugins/mssqlPlugin/src/main/java/com/external/plugins/MssqlPlugin.java:519]: JFYI, line no 496-500 add both username and password directly to HikariCP config, hence these lines are removed.

**Review by tkAppsmith — DISMISSED** [review]
Approving as this is removing redundant faulty code. Relying on QA regression to test against all supported mysql flavours.

File: app/server/appsmith-plugins/mssqlPlugin/src/test/java/com/external/plugins/MssqlTestDBContainerManager.java [hunk 2]
```diff
@@ -28,7 +28,7 @@ public static MSSQLServerContainer getMssqlDBForTest() {
                         .asCompatibleSubstituteFor("mcr.microsoft.com/mssql/server:2017-latest"))
                 .acceptLicense()
                 .withExposedPorts(1433)
-                .withPassword("Mssql123");
+                .withPassword("Mssql12;3");
```
Comment (tkAppsmith) [inline app/server/appsmith-plugins/mssqlPlugin/src/test/java/com/external/plugins/MssqlTestDBContainerManager.java:31]: Is this ok or typo??

Comment (sumitsum) [reply app/server/appsmith-plugins/mssqlPlugin/src/test/java/com/external/plugins/MssqlTestDBContainerManager.java:31]: Hi @tkAppsmith this is intentional. The issue that this PR fixes is wrt to handling of semicolon character in the password string.

**Review by tkAppsmith — APPROVED** [review]
Approving as approval was lost due to merge conflict

**Pull request merged**